### ISSUE TITLE
Add vagrantfile and ansible playbook for setting up POC demo cluster

### ISSUE THREAD

**Issue by mbukatov** [Tendrl/tendrl-ansible#6]
We need to add vagrant code for setup of POC (proof of concept) demonstration cluster.

This includes:

* [x] vagrantfile with definition of virtual machines for the demo cluster
* [ ] special ansible playbook tweaked for vagrant demo cluster use case
* [ ] make sure it works on Fedora with Vagrant on top of libvirt
* [ ] make sure it works on Mac OS X with Vagrant on top of VirtualBox or VMware (or whatever hypervisor available there, checking one is good enough, volunteer needed for this)

Having this work on Fedora with libvirt has higher priority over Mac OS X with proprietary hypervisors.

**Comment by mbukatov** [comment]
When starting the work on playbook for vagrant, I would suggest to check if `etcd_ip_address` variable could be set by using default ansible interface facts like this: `"{{ ansible_default_ipv4.address }}"` for *Tendrl Server* or `"{{ hostvars[groups['tendrl-server'][0]].ansible_default_ipv4.address }}"` for *Tendrl Storage Nodes*, and only if that doesn't not work, specify particular interface via `ansible_eth1.ipv4.address`.

**Comment by mbukatov** [comment]
Based on @japplewhite 's https://github.com/Tendrl/usmqe-setup/issues/76#issuecomment-309123355 we could also add that:

>  If you call the playbook as follows:
ansible-playbook -u vagrant -b site.yml
it will automatically overload the user and attempt to use sudo with no change in the playbooks. So there is no need to change from root. We just need to document that as a workaround.

**Comment by mbukatov** [comment]
Since tendrl-ansible doesn't contain any setup for gluster cluster (which is obviously out of scope for tendrl ansible, complicated and large task itself), this POC libvirt/vagrant demo setup is blocked by:

* https://github.com/Tendrl/node-agent/issues/467 (when Tendrl is installed, it's too late to create a gluster cluster for import)

**Comment by mbukatov** [comment]
There are some vagrant projects for POC Tendrl setup:

* https://github.com/shirshendu/multinode-glusterfs-with-tendrl-vagrant
* https://github.com/julienlim/multinode-glusterfs-with-tendrl-vagrant

See current work at https://github.com/shirshendu/multinode-glusterfs-with-tendrl-vagrant/pull/1

We may drop vagrant file from tendrl-ansible, which is not maintained anyway, and promote work of @julienlim and @shirshendu into Tendrl org on github when ready. Would that make sense?

**Comment by julienlim** [comment]
@mbukatov I suggested to @shirshendu that we start with https://github.com/red-hat-storage/RHGS-vagrant as Daniel Messer has done a nice job setting it up with minimal config for POC's.  It just needs to be tweaked to use tendrl.

**Comment by shirshendu** [comment]
That repo basically depends on custom boxes uploaded to a file server (they contain the previous release of tendrl); but apart from that one downside, there's a bunch of stuff setup through ansible, the way we need it for POCs. I hope to be depending on centos/7 instead, and adding dependencies. Better for the longer run.

It does have tendrl installation, but only the previous release. For running other versions (and git branches), we also need to add injection of code from the host into the right place in the vagrant machines. Latest release should be straightforward though, thanks to the tendrl-copr role.

My current need is only for a local dev-VM for tendrl-api, and i'll be structuring my work as such (i.e., not deploying the tendrl-server, and keeping etcd, tendrl-api, tendrl-ui, graphana etc on the host). Once that is done, let us adapt it to the general use-case for reliable VMs for testing.

**Comment by mbukatov** [comment]
Closing, as the vagrant setup will be maintained in a separate repository.
